src/middleware/builtin.ts

```diff
@@ -23,98 +17,110 @@ import {
   BlockElementAction,
   SlackViewAction,
   EventTypePattern,
-  ViewOutput,
 } from '../types';
 import { ActionConstraints, ViewConstraints, ShortcutConstraints, OptionsConstraints } from '../App';
 import { ContextMissingPropertyError } from '../errors';
 
+/** Type predicate that can narrow payloads block action or suggestion payloads */
+function isBlockPayload(
+  payload:
+  | SlackActionMiddlewareArgs['payload']
+  | SlackOptionsMiddlewareArgs['payload']
+  | SlackViewMiddlewareArgs['payload'],
+): payload is BlockElementAction | BlockSuggestion {
+  return 'action_id' in payload && payload.action_id !== undefined;
+}
+
+type CallbackIdentifiedBody =
+  | InteractiveMessage
+  | DialogSubmitAction
+  | MessageShortcut
+  | GlobalShortcut
+  | InteractiveMessageSuggestion
+  | DialogSuggestion;
+
+// TODO: consider exporting these type guards for use elsewhere within bolt
+// TODO: is there overlap with `function_executed` event here?
+function isCallbackIdentifiedBody(
+  body: SlackActionMiddlewareArgs['body'] | SlackOptionsMiddlewareArgs['body'] | SlackShortcutMiddlewareArgs['body'],
+): body is CallbackIdentifiedBody {
+  return 'callback_id' in body && body.callback_id !== undefined;
+}
+
+// TODO: clarify terminology used internally: event vs. body vs. payload
+/** Type predicate that can narrow event bodies to ones containing Views */
+function isViewBody(
+  body: SlackActionMiddlewareArgs['body'] | SlackOptionsMiddlewareArgs['body'] | SlackViewMiddlewareArgs['body'],
+): body is SlackViewAction {
+  return 'view' in body && body.view !== undefined;
+}
+
+function isEventArgs(
+  args: AnyMiddlewareArgs,
+): args is SlackEventMiddlewareArgs {
+  return 'event' in args && args.event !== undefined;
+}
+
+function isMessageEventArgs(
+  args: AnyMiddlewareArgs,
+): args is SlackEventMiddlewareArgs<'message'> {
+  return isEventArgs(args) && 'message' in args;
+}
+
 /**
  * Middleware that filters out any event that isn't an action
  */
-export const onlyActions: Middleware<AnyMiddlewareArgs & { action?: SlackAction }> = async (args) => {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const { action, next } = args as any; // FIXME: workaround for TypeScript 4.7 breaking changes
-  // Filter out any non-actions
-  if (action === undefined) {
-    return;
+export const onlyActions: Middleware<AnyMiddlewareArgs> = async (args) => {
```

Comment:
Breaking change: no longer need to call this as `onlyActions()`, instead, it is a middleware itself and can be simply passed as `onlyActions`. The wrapping closure within was unnecessary. This lines it up with other, non-parameterized middleware in this module.